### DISCUSSION TITLE
API to enable/disable discussions on a repository

### DISCUSSION THREAD

**Discussion by lfrancke** [community/community#36757] [Discussions] [answered]
I can't find a way to enable/disable discussions on a repository using the GraphQL API.

[`UpdateRepositoryInput`](https://docs.github.com/en/graphql/reference/input-objects#updaterepositoryinput) has `hasIssuesEnabled`, `hasProjectsEnabled` and `hasWikiEnabled` but no `hasDiscussionsEnabled`.

Is this hidden somewhere else or if not is this on the roadmap?

**Comment by Psimage** [comment]
I also can't find any REST API with ability to enable/disable discussions feature.
https://docs.github.com/en/rest/repos/repos#update-a-repository endpoint does not allow updating `has_discussions` field.

**Reply by LindseyB** [reply]
This is a recent addition, but you will see `has_discussions` is now available there.

**Comment by LangLangBart** [comment, marked as answer]
> I can't find a way to enable/disable discussions on a repository using the GraphQL API.


Praise the developers, they should join the community and take a victory lap instead of me posting a link to the changelog. :blush:

[Schema changes for 2022-11-05 - hasDiscussionsEnabled](https://docs.github.com/en/graphql/overview/changelog#schema-changes-for-2022-11-05)

```graphql
mutation {
  updateRepository(input: {hasDiscussionsEnabled: true, repositoryId: "MDEwOlJlcG9zaXRvcnkzMDE1NzMzNDQ="}) {
    repository {
      nameWithOwner
      hasDiscussionsEnabled
    }
  }
}
```

**Reply by lfrancke** [reply]
Awesome, thank you!

You beat me to it.

**Reply by sealor** [reply]
Thanks for sharing!

Unfortunately, this does not work for me. I get the following permission error:
```
mutation {
  updateRepository(input: {hasDiscussionsEnabled: true, repositoryId: "MDEwOlJlcG9zaXRvcnkyNzY0Mg=="}) {
    repository {
      nameWithOwner
      hasDiscussionsEnabled
    }
  }
}
```
```
GraphqlResponseError: Request failed due to following response errors:
 - fast-collaboration-github-app does not have permission to toggle discussions on FAST/github-app-test2.
```

We are using GitHub Enterprise v3.8 and I tried to grant several permissions (e.g. Administration and Discussions).
Any help is appreciated!

**Comment by muhammadnurulahsan** [comment]
```
mutation {
  updateRepository(input: {hasDiscussionsEnabled: true, repositoryId: "MDEwOlJlcG9zaXRvcnkzMDE1NzMzNDQ="}) {
    repository {
      nameWithOwner
      hasDiscussionsEnabled
    }
  }

}
```
